src/main/scala/com/microsoft/azure/cosmosdb/spark/rdd/CosmosDBRDDIterator.scala

```diff
@@ -197,12 +199,25 @@ class CosmosDBRDDIterator(hadoopConfig: mutable.Map[String, String],
         .getOrElse(FilterConverter.createQueryString(requiredColumns, filters))
       logInfo(s"CosmosDBRDDIterator::LazyReader, created query string: $queryString")
 
-      if (queryString == FilterConverter.defaultQuery) {
-        // If there is no filters, read feed should be used
-        connection.readDocuments(feedOpts)
-      } else {
-        connection.queryDocuments(queryString, feedOpts)
+      var iteratorDocument: Iterator[Document] = Iterator()
+
+      try {
+        if (queryString == FilterConverter.defaultQuery) {
+          // If there is no filters, read feed should be used
+          iteratorDocument = connection.readDocuments(feedOpts)
+        } else {
+          iteratorDocument = connection.queryDocuments(queryString, feedOpts)
+        }
+      }
+      catch {
+        case ex: IllegalStateException =>  logWarning(s"Received IllegalStateException PartitionKeyRangeId ${partition.partitionKeyRangeId} was split or gone");
```

Comment:
in addition to IllegalStateException split can cause DocumentClientException as Gone or also ServiceUnavailable, we should handle these as well right ?